pkg/scalers/cpu_memory_scaler.go

```diff
@@ -142,7 +210,294 @@ func (s *cpuMemoryScaler) GetMetricSpecForScaling(context.Context) []v2.MetricSp
 	return []v2.MetricSpec{metricSpec}
 }
 
-// GetMetricsAndActivity no need for cpu/memory scaler and always active for cpu/memory scaler
-func (s *cpuMemoryScaler) GetMetricsAndActivity(_ context.Context, _ string) ([]external_metrics.ExternalMetricValue, bool, error) {
-	return nil, true, nil
+func calculateAverage(total *resource.Quantity, count int64) *resource.Quantity {
+	if count == 0 {
+		return &resource.Quantity{}
+	}
+
+	// Convert the total to milli-units
+	nanoValue := total.ScaledValue(resource.Nano)
+
+	// Perform the division
+	averageNanoValue := nanoValue / count
+
+	// Create a new Quantity from the average milli-value
+	return resource.NewScaledQuantity(averageNanoValue, resource.Nano)
+}
+
+func (s *cpuMemoryScaler) getAverageValue(ctx context.Context, metricName string) (*resource.Quantity, error) {
```

Comment:
please add comment explaining this function (this applies to the whole PR where the function's function (:)) is not obvious